keg_storage/sftp.py

```diff
@@ -0,0 +1,134 @@
+import contextlib
+import logging
+
+from paramiko import SSHClient
+
+# from libs import crypto
+
+import keg_storage
+
+
+log = logging.getLogger(__name__)
+
+
+class BatchSFTPStorage(keg_storage.StorageBackend):
+    """
+    A `StorageBackend` that interacts with a single, pre-connected SFTP client.
+    """
+
+    def __init__(self, sftp_client, crypto_key=None, name='sftp'):
+        """
+        :param sftp_client: is an paramiko.SFTPClient object.
+        :param crypto_key: is a crypto key to use for encrypting files copied from the remote.
+        :param name: is the name of the storage backend.
+        """
+        self.sftp_client = sftp_client
+        self.crypto_key = crypto_key
+        self.name = name
+
+    def list(self, path):
+        return self.sftp_client.listdir(path)
+
+    def get(self, path, dest):
+        if self.crypto_key:
```

Comment:
@rsyring, having a key set and wanting something decrypted are two different things. 

Since these objects live at a global level the `crypto_key` isn't a great indicator of intention. For example, lets say you want to get an encrypted file at path `/secret.zip.enc` from the server,  then you want to get `/logs/2017.01.02.log` which isn't encrypted, you can't do that using just the `get` method. You will have to know the implementation of this particular backend, which is, if the `crypto_key` is set it will try to decrypt it. That is behavior isn't obvious and thus I would suggest we change it.

That being said, the direction I would like to see this go is with #2. Encryption becomes an optional, though baked in part, of every implementation.